src/r2mm/downloading/BetterThunderstoreDownloader.ts

```diff
@@ -14,66 +13,36 @@ import { ImmutableProfile } from '../../model/Profile';
-                    let otherVersionAlreadyAdded = false;
-                    builder.forEach(v => {
-                        // If otherVersionAlreadyAdded, or full names are equal
-                        otherVersionAlreadyAdded = otherVersionAlreadyAdded || v.getMod().getFullName() === matchingProvider.getFullName();
-                    });
-                    if (!otherVersionAlreadyAdded) {
-                        const tsCombo = new ThunderstoreCombo();
-                        tsCombo.setMod(matchingProvider);
-                        tsCombo.setVersion(matchingVersion);
-                        foundDependencies.push(tsCombo);
-                    }
-                }
-            }
-        })
-        foundDependencies.forEach(found => builder.push(found));
-        foundDependencies.forEach(found => this.buildDependencySet(found.getVersion(), allMods, builder));
-        return builder;
+    public async buildDependencySet(mod: ThunderstoreVersion, builder: ThunderstoreCombo[]): Promise<void> {
+        await this._buildDependencySet(mod, builder);
     }
 
-    public buildDependencySetUsingLatest(mod: ThunderstoreVersion, allMods: ThunderstoreMod[], builder: ThunderstoreCombo[]): ThunderstoreCombo[] {
-        const foundDependencies = new Array<ThunderstoreCombo>();
-        mod.getDependencies().forEach(dependency => {
-            // Find matching ThunderstoreMod.
-            const matchingProvider: ThunderstoreMod | undefined = allMods.find(o => dependency.startsWith(o.getFullName() + "-"));
-            if (matchingProvider !== undefined) {
-                // Get latest version of dependency
-                const matchingVersion = matchingProvider.getVersions().reduce((one: ThunderstoreVersion, two: ThunderstoreVersion) => {
-                    if (one.getVersionNumber().isNewerThan(two.getVersionNumber())) {
-                        return one;
-                    } return two;
-                });
-                let otherVersionAlreadyAdded = false;
-                builder.forEach(v => {
-                    // If otherVersionAlreadyAdded, or full names are equal
-                    otherVersionAlreadyAdded = otherVersionAlreadyAdded || v.getMod().getFullName() === matchingProvider.getFullName();
-                });
-                if (!otherVersionAlreadyAdded) {
-                    const tsCombo = new ThunderstoreCombo();
-                    tsCombo.setMod(matchingProvider);
-                    tsCombo.setVersion(matchingVersion);
-                    foundDependencies.push(tsCombo);
-                }
-            }
-        })
+    public async buildDependencySetUsingLatest(mod: ThunderstoreVersion, builder: ThunderstoreCombo[]): Promise<void> {
+        const useLatestVersion = true;
+        await this._buildDependencySet(mod, builder, useLatestVersion);
+    }
+
+    private async _buildDependencySet(mod: ThunderstoreVersion, builder: ThunderstoreCombo[], useLatestVersion=false): Promise<void> {
```

Comment:
This is more of a preference thing but it would be nicer to not have this defaulted. It could be more readable to create an Enum and we can utilise `DependencySetBuilderStyle.USE_LATEST` as a more readable equivalent wherever needed.

Default behaviour is weird IMO and we'll just prevent issues by being explicit.

Reply:
Generally speaking I don't consider parameter default values a negative thing. Of course one can go overboard with them, and e.g. using mutable default values is a trap. Constants and enums also have their use, especially to avoid typos on magic strings. I'm not sure how well that applies to booleans though.

In this particular case, we're talking about a private method, which is only called by the two named wrapper methods above. So I'd argue that having the named methods is more readable than having _buildDependencySet use enum parameter. (Also the named methods are public although they're only used internally, so to get rid of them would require changing the interface. Which might not be a bad thing in itself, but is a bit of scope creep.)

Reply:
The problem with default values is that it's extremely easy to miss them out. I'd much rather we were explicit with our signatures. 

A boolean is IMO still a magic variable.
If you were to see the following on its own with no other context:
```ts
_buildDependencySet(mod, dependencies, true)
```
The `true` value doesn't really mean anything.

Equally, if you were to see:
```ts
_buildDependencySet(mod, dependencies, true)
...
_buildDependencySet(mod, dependencies)
...
_buildDependencySet(mod, dependencies, false)
```
You don't really know what's being done differently without context of the parameter's name